packages/actor-rdf-source-identifier-sparql/lib/ActorRdfSourceIdentifierSparql.ts

```diff
@@ -0,0 +1,43 @@
+import {IActionHttp, IActorHttpOutput} from "@comunica/bus-http";
+import {ActorRdfSourceIdentifier, IActionRdfSourceIdentifier,
+  IActorRdfSourceIdentifierOutput} from "@comunica/bus-rdf-source-identifier";
+import {Actor, IActorArgs, IActorTest, Mediator} from "@comunica/core";
+
+/**
+ * A comunica SPARQL RDF Source Identifier Actor.
+ */
+export class ActorRdfSourceIdentifierSparql extends ActorRdfSourceIdentifier {
+
+  public readonly mediatorHttp: Mediator<Actor<IActionHttp, IActorTest, IActorHttpOutput>,
+    IActionHttp, IActorTest, IActorHttpOutput>;
+
+  constructor(args: IActorRdfSourceIdentifierSparqlArgs) {
+    super(args);
+  }
+
+  public async test(action: IActionRdfSourceIdentifier): Promise<IActorTest> {
+    if (!action.sourceValue.startsWith('http')) {
+      throw new Error(`Actor ${this.name} can only detect SPARQL endpoints hosted via HTTP(S).`);
+    }
+    const url: string = action.sourceValue + '?query=' + encodeURIComponent('ASK { ?s a ?o }');
+    const headers: Headers = new Headers();
+    headers.append('Accept', 'application/sparql-results+json');
+    const httpAction: IActionHttp = { input: url, init: { headers, method: 'HEAD' } };
+    const httpResponse: IActorHttpOutput = await this.mediatorHttp.mediate(httpAction);
+    if (!httpResponse.ok || httpResponse.headers.get('Content-Type').indexOf('application/sparql-results+json') < 0) {
```

Comment:
Do all sparql endpoints support sparql json?

Reply:
This is not a requirement in the spec. But in practise that's the go-to format. (And again, it's heuristic :-) )